commons/src/main/java/fr/free/nrw/commons/upload/ImageProcessing.java

```diff
@@ -0,0 +1,127 @@
+package fr.free.nrw.commons.upload;
+
+import android.content.Context;
+import android.database.Cursor;
+import android.media.ExifInterface;
+import android.net.Uri;
+import android.provider.DocumentsContract;
+import android.provider.MediaStore;
+import android.util.Log;
+
+import java.io.IOException;
+
+
+/**
+ * Created by misao on 16-Dec-15.
```

Comment:
This is controversial, and I don't what @nicolas-raoul's code conventions are, but we don't need @author tags or block comments. We can use Git history to see who did what. I know the IDE adds these by default which lends opposition to my opinion. Clean code, in my opinion, is just code as much as possible.